Sources/SPTPersistentCacheGarbageCollectorScheduler.m

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2016 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#import "SPTPersistentCacheGarbageCollectorScheduler.h"
+
+#import "SPTPersistentCache+Private.h"
+
+const NSTimeInterval SPTPersistentCacheTimerProxyTimerToleranceInterval = 300;
+
+@interface SPTPersistentCacheGarbageCollectorScheduler ()
+@property (nonatomic, copy) SPTDataCacheDebugCallback debugOutput;
+@property (nonatomic, strong) NSTimer *timer;
+@property (nonatomic, strong) SPTPersistentCacheOptions *options;
+@end
+
+
+@implementation SPTPersistentCacheGarbageCollectorScheduler
+
+#pragma mark - Initializer
+
+- (instancetype)initWithDataCache:(SPTPersistentCache *)dataCache
+                          options:(SPTPersistentCacheOptions *)options
+                            queue:(dispatch_queue_t)queue
+{
+    if (!(self = [super init])) {
+        return nil;
+    }
+    
+    _options = options;
+    _dataCache = dataCache;
+    _queue = queue;
+    _debugOutput = [options.debugOutput copy];
+    
+    return self;
+}
+
+- (void)dealloc
+{
+    NSTimer *timer = self.timer;
```

Comment:
Also the timer will retain this object, making the timer never valid when it gets to dealloc, its better off to have an "invalidate" method on this object which calls invalidate on the timer